code-challenges.test.js

```diff
@@ -21,36 +20,146 @@
+// HOF logic:
+//      Create this conditional: typeof value = "number" && value % 2 !== 0
+// Have oddArray be equal to oddArray.sort((a, b) => a - b)
+// return oddArray
+// Run the test code to make sure it passes green
+
+const oddSorter = (arr) => {
+  let oddArray = arr
+    .filter((value) => typeof value === "number" && value % 2 !== 0)
+    .sort((a, b) => a - b);
+  return oddArray;
+};
+
+// Notes: I ran the initial test and it failed red with the right error (function not defined). I created the code and ran the test again, got it passing green. I did some minor refactoring. I took the variable declaration and combined it with the .filter and the .sort, so it's all in one line. In terms of methodology, .filter is the way to go here.
 
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var numbersToAdd1 = [2, 4, 45, 9]
+var numbersToAdd1 = [2, 4, 45, 9];
 // Excpected output: [2, 6, 51, 60]
 
-var numbersToAdd2 = [0, 7, -8, 12]
+var numbersToAdd2 = [0, 7, -8, 12];
 // Expected output: [0, 7, -1, 11]
 
-var numbersToAdd3 = []
+var numbersToAdd3 = [];
 // Expected output: []
 
+// 3a) Pseudo code:
+// Copy and paste test code template into assignment (make sure nothing is inside initial template)
+// Read problem statement and understand inputs/outputs (inputs need to be directly inserted into test code, and not as variables)
+// Create relevant info for describe, it, expect, and toEqual methods
+// Test code output to ensure it is failing red (should get a reference error saying function is not defined)
+
+describe("accumulator", () => {
+  it("a function which takes in an array and returns another array, with the values in each element being a cumulative sum of previous values from the input array", () => {
+    expect(accumulator([2, 4, 45, 9])).toEqual([2, 6, 51, 60]);
+    expect(accumulator([0, 7, -8, 12])).toEqual([0, 7, -1, 11]);
+    expect(accumulator([])).toEqual([]);
+  });
+});
 
 // b) Create the function that makes the test pass.
+
+// 3b) Pseudo code:
+// Declare a function and call it accumulator (const declaration)
+// Name the parameter as arr
+// Declare a variable named accumulatorArray, and initialize it as an empty array (let declaration)
+// Declare another variable, name it sum, and initialize it with a value of 0 (let declaration)
+// Create an if statement for returning an empty array if an empty array is the input
+// if statement logic:
+//      if arr === [], return accumulatorArray
+// Apply a .map HOF to accumulatorArray
+// .map logic:
+//      sum + value
+// return accumulatorArray
+
+const accumulator = (arr) => {
+  let accumulatorArray = [];
+  let sum = 0;
+  if (arr === []) {
+    return accumulatorArray;
+  }
+  accumulatorArray = arr.map((value) => (sum += value));
+  return accumulatorArray;
+};
+
```

Comment:
Yeah this looks great. I think the problem of leaving the sum without a value is that you would then try and perform addition on the data type undefined.